service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterMapper.java

```diff
@@ -53,10 +61,38 @@ public String mapEncounterToEhrComposition(Encounter encounter) {
             .displayName(buildDisplayName(encounter))
             .originalText(buildOriginalText(encounter));
 
+        final Optional<String> recReference = findParticipantWithCoding(encounter, ParticipantCoding.RECORDER)
+            .map(idMapper::get);
+
+        recReference.ifPresent(encounterStatementTemplateParameters::author);
```

Comment:
> Exception - if there is no participant of type code REC then log the issue and stop process it's a MUST as far as GPC spec is concerned

Not optional, we need to throw an error if we don't find the RECORDER participant.